src/map/atcommand.c

```diff
@@ -1887,6 +1887,63 @@ ACMD(hair_color)
 	return true;
 }
 
+ACMD(setzone)
+{
+	char zone_name[MAP_ZONE_MAPFLAG_LENGTH];
+	memset(zone_name, '\0', sizeof(zone_name));
+
+	char fmt_str[8] = "";
+	snprintf(fmt_str, 8, "%%%ds", MAP_ZONE_MAPFLAG_LENGTH - 1);
+
+	if (*message == '\0' || sscanf(message, fmt_str, zone_name) < 1) {
+		clif->message(fd, msg_fd(fd, 924)); // usage info
+		return false;
+	}
+
+	struct map_zone_data *zone = strdb_get(map->zone_db, zone_name);
+	const char *prev_zone_name = map->list[sd->bl.m].zone->name;
```

Comment:
I think ``map->list[sd->bl.m].zone`` can be NULL if zone "all" not present in map_zone_db.conf

Reply:
no, unlike custom zones `All` is in `map->zone_all` rather than in `map->zone_db` so even if it's missing from the zone db it won't be a null pointer and thus it won't break this command